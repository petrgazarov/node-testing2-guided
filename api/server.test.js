const db = require('../data/dbConfig')
const request = require('supertest')
const server = require('./server')

const listOfHobbits = [{ name: 'sam' }, { name: 'frodo' }]

beforeAll(async () => {
  await db.migrate.rollback() // takes back the db to the stone age
  await db.migrate.latest() // bringing to the latest
})
beforeEach(async () => {
  await db('hobbits').truncate() // wipes rows, and resets id numbers
})
afterAll(async () => {
  await db.destroy()
})

describe('[GET] /hobbits', () => {
  beforeEach(async () => {
    await db('hobbits').insert(listOfHobbits)
  })
  it('responds with a 200 OK', async () => {
    const res = await request(server).get('/hobbits')
    expect(res.status).toBe(200)
  })
  it('returns a list of hobbits', async () => {
    const res = await request(server).get('/hobbits')
    console.log(res.body)
    console.log(res.status)
    console.log(res.headers)
    expect(res.body).toMatchObject(listOfHobbits)
    expect(res.body[0]).toMatchObject({ name: 'sam' })
    expect(res.body[0]).toHaveProperty('name', 'sam')

    expect(res.body).toMatchSnapshot()
    // const res = request(server).post('/hobbits').send({ name: 'pippin' })
  })
})
