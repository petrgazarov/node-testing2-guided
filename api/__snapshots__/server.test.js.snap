// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[GET] /hobbits returns a list of hobbits 1`] = `
Array [
  Object {
    "id": 1,
    "name": "sam",
  },
  Object {
    "id": 2,
    "name": "frodo",
  },
]
`;
